tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalCovectorAndMagnitude.cpp

```diff
@@ -0,0 +1,298 @@
+      internal_directions, domain::Tags::UnnormalizedFaceNormal<Dim>>>(*box);
+  for (const auto& [direction, neighbors] :
+       db::get<domain::Tags::Element<Dim>>(*box).neighbors()) {
+    (void)neighbors;
+    Variables<tmpl::conditional_t<
+        UseFlatSpace,
+        tmpl::list<
+            evolution::dg::Actions::detail::OneOverNormalVectorMagnitude>,
+        tmpl::list<evolution::dg::Actions::detail::OneOverNormalVectorMagnitude,
+                   evolution::dg::Actions::detail::NormalVector<Dim>,
+                   InverseSpatialMetric<Dim>>>>
+        fields_on_face{
+            db::get<domain::Tags::Interface<internal_directions,
+                                            domain::Tags::Mesh<Dim - 1>>>(*box)
+                .at(direction)
+                .number_of_grid_points()};
+    if constexpr (not UseFlatSpace) {
+      double temp = 0.1;
+      for (auto& component : get<InverseSpatialMetric<Dim>>(fields_on_face)) {
+        for (double& t : component) {
+          t = temp;
+          temp *= 1.1;
+        }
+      }
+    }
+
+    evolution::dg::Actions::detail::
+        unit_normal_vector_and_covector_and_magnitude<tmpl::conditional_t<
+            UseFlatSpace, FlatSpaceSystem, SystemWithInverseMetric<Dim>>>(
+            box, make_not_null(&fields_on_face), direction,
+            unnormalized_normal_covectors);
+    if constexpr (UseFlatSpace) {
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          get(magnitude(
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorAndMagnitude<Dim>>(*box)
+                       .at(direction)))));
+    } else {
+      const auto& normal_covector =
+          get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+              *get<evolution::dg::Tags::InternalFace::
+                       NormalCovectorAndMagnitude<Dim>>(*box)
+                   .at(direction));
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          sqrt(get(
+              dot_product(normal_covector, normal_covector,
+                          get<InverseSpatialMetric<Dim>>(fields_on_face)))));
+    }
+
+    for (const double magnitude_of_normal :
+         get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+             *get<evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<
+                  Dim>>(*box)
+                  .at(direction)))) {
+      CHECK(magnitude_of_normal > 0.0);
+    }
+    for (size_t i = 0; i < Dim; ++i) {
+      CHECK_ITERABLE_APPROX(
+          unnormalized_normal_covectors.at(direction).get(i),
+          DataVector{
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
```

Comment:
this line would be easier to read if the get from the box was on a separate line:
```
const auto& normal_covector_and_magnitude = 
  *get<evolution::dg::Tags::InternalFace::
          NormalCovectorAndMagnitude<Dim>>(*box)
      .at(direction);
```